sdk/azcore/runtime/request.go

```diff
@@ -161,13 +198,32 @@ func SetMultipartFormData(req *policy.Request, formData map[string]any) error {
 				}
 			}
 			continue
+		} else if mpc, ok := v.(streaming.MultipartContent); ok {
+			if err := writeMultipartContent(k, k, mpc); err != nil {
+				return err
+			}
+			continue
+		} else if mpcs, ok := v.([]streaming.MultipartContent); ok {
+			for _, mpc := range mpcs {
+				if err := writeMultipartContent(k, k, mpc); err != nil {
+					return err
+				}
+			}
+			continue
 		}
-		// ensure the value is in string format
-		s, ok := v.(string)
-		if !ok {
-			s = fmt.Sprintf("%v", v)
+
+		var content string
+		if b, ok := v.([]byte); ok {
```

Comment:
Just FYI in case you imagine we'll ever use `gocritic`, I think it wants `switch` instead of if-else if-else

Reply:
Converted to use a type switch.